### ISSUE TITLE
Compose fails to package explicit artifacts

### ISSUE THREAD

**Issue by danielvijge** [serverless/serverless#12824]
### Issue description

When using compose on services that have explicit artifact packages, the command fails with `✖ Error: Error: ENOENT: no such file or directory, open 'index.js' encountered during hash calculation for provided filePath: index.js`

Steps to reproduce:
1. Create two services, named `service-a` and `service-b`. I used the starter templates for both. Only add `package.artifact` to each service. Both services are the same, except for the service name.
```yaml
service: service-a
frameworkVersion: '4'

provider:
  name: aws
  runtime: nodejs18.x

functions:
  function1:
    handler: index.handler
    package:
      artifact: index.js
```
3. Create a compose file for both services.
```yaml
services:
  service-a:
    path: service-a
  service-b:
    path: service-b
```
4. Try to run a compose command, e.g. `serverless package`. This fails with the message `✖ Error: Error: ENOENT: no such file or directory, open 'index.js' encountered during hash calculation for provided filePath: index.js`

Note if the files really do no exist at all, e.g. by renaming `handler.js` files, there is a different error message: `✖ Cannot access package artifact at "index.js" (for "function1"): ENOENT: no such file or directory, access '/code/service-b/index.js'`

Going into a service folder and trying to package a service individually (`cd service-a; serverless package`)

### Context

Serverless ϟ Support

This generates a report from your last Serverless Framework command (including any errors) to use for Github Issues, debugging w/ AI, or creating a support ticket w/ Serverless Inc.

Summary Report -----------------

# Service Overview
- Serverless Framework Version: 4.4.0
- Service Path: /code
- Service Config File: serverless-compose.yaml
- Command: package 

## Service Config
```
services:
  service-a:
    path: service-a
  service-b:
    path: service-b
```

# Service Overview
- Serverless Framework Version: 4.4.0
- Service Path: /code/service-a
- Service Config File: serverless.yml
- Service Name: service-a
- Service Runtime: nodejs18.x
- Service Stage: dev
- Service Region: us-east-1
- Command: package 

## Error Message
Error: Error: ENOENT: no such file or directory, open 'index.js' encountered during hash calculation for provided filePath: index.js

## Error Stacktrace
```
at ReadStream.<anonymous> (file:///root/.serverless/releases/4.4.0/package/dist/sf-core.js:735:41032)
at ReadStream.emit (node:events:519:28)
at ReadStream.emit (node:domain:488:12)
at emitErrorNT (node:internal/streams/destroy:169:8)
at emitErrorCloseNT (node:internal/streams/destroy:128:3)
at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

## Service Config
```
service: service-a
frameworkVersion: '4'
provider:
  name: aws
  runtime: nodejs18.x
functions:
  function1:
    handler: index.handler
    package:
      artifact: index.js
```

# Service Overview
- Serverless Framework Version: 4.4.0
- Service Path: /code/service-b
- Service Config File: serverless.yml
- Service Name: service-b
- Service Runtime: nodejs18.x
- Service Stage: dev
- Service Region: us-east-1
- Command: package 

## Error Message
Error: Error: ENOENT: no such file or directory, open 'index.js' encountered during hash calculation for provided filePath: index.js

## Error Stacktrace
```
at ReadStream.<anonymous> (file:///root/.serverless/releases/4.4.0/package/dist/sf-core.js:735:41032)
at ReadStream.emit (node:events:519:28)
at ReadStream.emit (node:domain:488:12)
at emitErrorNT (node:internal/streams/destroy:169:8)
at emitErrorCloseNT (node:internal/streams/destroy:128:3)
at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

## Service Config
```
service: service-b
frameworkVersion: '4'
provider:
  name: aws
  runtime: nodejs18.x
functions:
  function1:
    handler: index.handler
    package:
      artifact: index.js
```

# Service Overview
- Serverless Framework Version: 4.4.0
- Command: support --summary true

Above is a summary report of your last Serverless Framework command.

Use it to debug with AI (e.g. ChatGPT) or report a new Github Issue: https://slss.io/issue

**Comment by czubocha** [comment]
Hey @danielvijge, thanks for reporting this. We’ve identified the issue and will be releasing a fix soon.

**Comment by czubocha** [comment]
@danielvijge the fix has been shipped in [v4.4.5](https://github.com/serverless/serverless/releases/tag/v4.4.5)
